gk/main.c

```diff
@@ -1727,11 +1729,51 @@ process_pkts_front(uint16_t port_front, uint16_t port_back,
         * IPv4: 14 + 8 + 20 = 42
         * IPv6: 14 + 8 + 40 = 62
         */
-       for (i = 0; i < num_rx; i++)
-               rte_prefetch0(rte_pktmbuf_mtod_offset(rx_bufs[i], void *, 0));
+	for (i = 0; i < PREFETCH_OFFSET && i < num_rx; i++)
+		rte_prefetch0(rte_pktmbuf_mtod_offset(rx_bufs[i], void *, 0));
 
 	/* Extract packet and flow information. */
-	for (i = 0; i < num_rx; i++) {
+	for (i = 0; i < (num_rx - PREFETCH_OFFSET); i++) {
+		struct ipacket *packet = &pkt_arr[num_ip_flows];
+		struct rte_mbuf *pkt = rx_bufs[i];
+
+		rte_prefetch0(rte_pktmbuf_mtod_offset(
+			rx_bufs[i + PREFETCH_OFFSET], void *, 0));
```

Comment:
The only difference between the body of this `for` loop and the following `for` loop is the prefetch above. Write a function for the body of this `for` loop without the prefetch and avoid the duplicate code. If measurements point that having a function here presents a performance issue, make that function inline.

Reply:
After evaluating the updated patch, there is no performance issue.